### PR TITLE
add a filter function that filters a list of results and updates the …

### PR DESCRIPTION
**Pull request by binhtendo**
…filtered list in the local state.
based on the search term, you can now enter locations you already know in the filter search field and favorite them.

### CODE REVIEW

File: components/Filter/index.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import React, { useState } from "react";
+
+const Filter = ({ results, setFilteredResults }) => {
+  const [filterText, setFilterText] = useState("");
+
+  const handleFilterChange = (e) => {
+    const newFilterText = e.target.value;
```
Comment (PReichetanz) [inline components/Filter/index.js:7]: There is no need to be brief here, you can write `event` to make clear what's happening.

File: components/Filter/index.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    setFilterText(newFilterText);
+
+    const filteredResults = results.filter((result) =>
+      result.name.toLowerCase().includes(newFilterText.toLowerCase())
+    );
+    setFilteredResults(filteredResults);
+  };
+
+  return (
+    <div>
```
Comment (PReichetanz) [inline components/Filter/index.js:17]: I am not sure here, but the `input` without a wrapping `form` element feels weird.

File: components/Searchbox/index.js [hunk 3]
```diff
@@ -1,38 +1,44 @@
...
       const places = await fetchPlaces(coordinates.lat, coordinates.lon);
       setResults(places);
+      setFilteredResults(places);
     } else {
       const errorMessage = "Eintrag konnte nicht gefunden werden.";
       window.alert(errorMessage);
     }
   };
+
   console.log(favorites);
```
Comment (PReichetanz) [inline components/Searchbox/index.js:34]: Please don't commit console.logs.

File: components/Searchbox/index.js [hunk 4]
```diff
@@ -1,38 +1,44 @@
...
   return (
     <div>
```
Comment (PReichetanz) [inline components/Searchbox/index.js:36]: Do you really need this `div` without any styling?

File: components/Searchbox/index.js [hunk 5]
```diff
@@ -1,38 +1,44 @@
...
       <form onSubmit={handleSubmit}>
         <label htmlFor="searchbox"></label>
         <input
           type="text"
-          value={query}
+          value={query || ""}
```
Comment (PReichetanz) [inline components/Searchbox/index.js:41]: Since `query` is always at least an empty string, do you really need this?

**Review by PReichetanz — COMMENTED** [review]
Nice work! I've found some issues to discuss 🚀

**Review by bk1405bgl — APPROVED** [review]
well done

**Pull request merged**